sympy/physics/mechanics/body.py

```diff
@@ -136,6 +136,25 @@ def __init__(self, name, masscenter=None, mass=None, frame=None,
         else:
             RigidBody.__init__(self, name, masscenter, frame, _mass, _inertia)
 
+        self._x = self.frame.x
+        self._y = self.frame.y
+        self._z = self.frame.z
+
+    @property
+    def x(self):
+        """The basis Vector for the Body, in the x direction. """
+        return self._x
```

Comment:
```suggestion
        return self,frame,x
```

Reply:
No reason not to do this, and it would ensure it always returns the correct value.

Reply:
I have declared `self._x` to `self.frame.x` in line 139